shell/pages/c/_cluster/settings/index.vue

```diff
@@ -3,13 +3,13 @@ import { NAME as SETTINGS } from '@shell/config/product/settings';
 import { MANAGEMENT } from '@shell/config/types';
 
 export default {
-  middleware({ redirect, route, store } ) {
-    const hasSettings = !!store.getters[`management/schemaFor`](MANAGEMENT.SETTING);
+  beforeCreate() {
```

Comment:
I'll preface this comment by saying that I'm fine with the approach represented here, I'm more interested in exploring an alternative to see if it makes sense for this use case. 

Have we considered utilizing in-component guards[^1] instead of the `beforeCreate()` hook? I think that the guard would better communicate what we are doing within this component, but it comes with the tradeoff that we might not have access to `this` when we want it.. 

[^1]: https://v3.router.vuejs.org/guide/advanced/navigation-guards.html#in-component-guards

Reply:
I haven't considered that. Do you know of a way to access the store from the guard? 

It's actually a pretty reasonable idea which may apply better in that it executes before the component is created just like middleware.

Reply:
I did find https://github.com/vuejs/vue-router/issues/1103#issuecomment-938511312

Reply:
Yeah, it looks like we would have to do something like this instead

```
beforeRouteEnter (to, from, next) {
  next(vm => {
    const hasSettings = !!vm.$store.getters[`management/schemaFor`](MANAGEMENT.SETTING);

    ...
  })
}
```

Reply:
I'll keep looking into this. It's interesting and not as straight forward as I would've thought. If we access the vm it looks like the component will already be created so not a big advantage.

![image](https://github.com/rancher/dashboard/assets/55104481/ea2c04d0-79c9-40bd-982a-b74b540ff9ba)

There's another suggestion I found which mentions just importing our store and accessing it directly. With responses that would resemble our current predicament in regards to how we can access our store in the app.

![image](https://github.com/rancher/dashboard/assets/55104481/11677d48-73a5-4482-9113-9a7011ecc70a)

Any thoughts on the idea of making the store global?

Reply:
I gave the global store a shot and it's a no go. The schemas aren't present when re/loaded.

I'm fine with leaving it as is or using the next(vm => {}). Do you think one is better than the other?

Reply:
I think we can leave this as is. It sounds like we won't get much of a benefit and the syntax will end up being less direct with the in-component guard.